ROME 1.6 seems to not work on Android

I recently started to use ROME and jumped directly to 1.6. Therefore, when trying to get a feed I always get the same error
`com.rometools.rome.io.ParsingFeedException: Invalid XML
[...]
Caused by: org.jdom2.JDOMException: http://apache.org/xml/features/disallow-doctype-decl feature not recognized for SAX driver org.apache.harmony.xml.ExpatReader`

I used some sample codes found on the web, but all are using older versions of ROME tools. As they actually, I tried to use the same RSS feeds they use, I've got the same error. Note that every working example that I found used FeedFetcher.

--- a/rome/src/main/java/com/rometools/rome/io/WireFeedInput.java
+++ b/rome/src/main/java/com/rometools/rome/io/WireFeedInput.java
@@ -359,23 +359,29 @@ public class WireFeedInput {
             }
 
         } catch (final JDOMException e) {
-            throw new IllegalStateException("JDOM could not create a SAX parser");
+            throw new IllegalStateException("JDOM could not create a SAX parser", e);
         }
         
         saxBuilder.setExpandEntities(false);
-        
+
         return saxBuilder;
-        
+
     }
     
     private void setFeature(SAXBuilder saxBuilder, XMLReader parser, String feature, boolean value) {
-        try {
+        if (isFeatureSupported(parser, feature, value)) {
             saxBuilder.setFeature(feature, value);
+        }
+    }
+
+    private boolean isFeatureSupported(XMLReader parser, String feature, boolean value) {
+        try {
             parser.setFeature(feature, value);
+            return true;
         } catch (final SAXNotRecognizedException e) {
-            // ignore
+            return false;
         } catch (final SAXNotSupportedException e) {
-            // ignore
+            return false;
         }
     }
 

